gcs/blockcf2/maxsize_test.go

```diff
@@ -0,0 +1,268 @@
+// the values change, this test breaks and any code that relies on assumptions
+// about the max filter size having a particular value are reviewed.
+func TestMaxSize(t *testing.T) {
+
+	// entrySizeForSize returns the size of scripts that ensures that every
+	// entry on a filter may be unique (i.e. no two scripts will generate
+	// the same internal value after siphash and modulo reductions).
+	entrySizeForSize := func(size uint32) uint32 {
+		switch {
+		case size < 1<<8:
+			return 1
+		case size < 1<<16:
+			return 2
+		case size < 1<<24:
+			return 3
+		default:
+			return 4
+		}
+	}
+
+	// mainnetMaxBlockSize is the max size of mainnet blocks according to
+	// the most recent consensus rules.
+	mainnetBlockSizes := chaincfg.MainNetParams().MaximumBlockSizes
+	mainnetMaxBlockSize := uint32(mainnetBlockSizes[len(mainnetBlockSizes)-1])
+
+	// minTxSize is the minimum transaction size with one input (and no outputs).
+	minTxSize := uint32(12 + 57 + 3 + 2)
+
+	// txOutSize is the size of an output, plus one byte for the pkscript
+	// length encoded as a varint.
+	txOutSize := uint32(10 + 1)
+
+	// p2shSize is the size of a P2SH script (the smallest standard script
+	// that may be added without limits to a transaction).
+	p2shSize := uint32(23)
+
+	tests := []struct {
+		name string
+		n    uint32
+		want uint64
+	}{{
+		// This test asserts the size of a filter when the source
+		// data for the filter fills an entire P2P message.
+		name: "filter from data that fills an entire P2P msg",
+		n:    wire.MaxMessagePayload / entrySizeForSize(wire.MaxMessagePayload),
+		want: 22541387,
+	}, {
+		// This test asserts the size of a filter when the source data
+		// for the filter fills as many bytes as the maximum block
+		// payload size for a P2P message.
+		name: "filter from data that fills the max block size",
+		n:    wire.MaxBlockPayload / entrySizeForSize(wire.MaxBlockPayload),
+		want: 1174034,
+	}, {
+		// This test asserts the size of a filter when the source data
+		// for the filter fills as many bytes as the maximum consensus
+		// enforced block size for mainnet.
+		name: "filter from data that fills the max mainnet block size",
+		n:    mainnetMaxBlockSize / entrySizeForSize(mainnetMaxBlockSize),
+		want: 352214,
+	}, {
+		// This test asserts the size of a filter when the source data
+		// for the filter is a tx that has as many OP_RETURN outputs
+		// as necessary to fill a block.
+		name: "filter from tx filled with OP_RETURN outputs",
+		n:    wire.MaxBlockPayload / (11 + entrySizeForSize(wire.MaxBlockPayload)),
+		want: 251581,
+	}, {
+		// This test asserts the size of a filter when the source data
+		// is a set of transactions that have 4 OP_RETURN outputs, as
+		// enforced by the standardness policy checks of the mempool.
+		name: "filter from standard OP_RETURN tx",
+		n:    wire.MaxBlockPayload / (minTxSize + (txOutSize+entrySizeForSize(wire.MaxBlockPayload))*4) * 4,
```

Comment:
Perhaps I'm missing something, but this doesn't look correct to me.  It'll do the division and then multiply by 4.  I believe it should not have that final multiplication by 4.

That would make n = 1310720 / (73 + 56) = 10160 versus the current 40640.